### PR TITLE
Bump Z3 version in static Z3 PPA build to 4.8.9.

### PR DESCRIPTION
**Pull request by ekpyron**
First step in a series of PRs to update the Z3 version everywhere.
I haven't run the script yet, so please ping me, once we're clear that we go through with this before the next release, then I'll run it and link the build status here :-).

Should be the last merged.
Depends on https://github.com/ethereum/solidity/pull/9797

### CODE REVIEW

File: scripts/deps-ppa/static_z3.sh [hunk 1]
```diff
@@ -80,7 +80,9 @@ Vcs-Browser: https://github.com/Z3Prover/z3
 
 Package: libz3-static-dev
 Section: libdevel
-Architecture: any-i386 any-amd64
+Architecture: any-amd64
```
Comment (ekpyron) [inline scripts/deps-ppa/static_z3.sh:83]: We did change this in the release PPA script and our builds fail for i386 ubuntu, so we can probably drop support here as well.

File: scripts/deps-ppa/static_z3.sh [hunk 2]
```diff
@@ -80,7 +80,9 @@ Vcs-Browser: https://github.com/Z3Prover/z3
...
+Breaks: libz3-dev
+Replaces: libz3-dev
```
Comment (ekpyron) [inline scripts/deps-ppa/static_z3.sh:85]: @Marenz requested those - not sure it works like this without version, but we can try and see if it works.
Ref: https://www.debian.org/doc/debian-policy/ch-relationships.html#packages-which-break-other-packages-breaks - but I only quickly read through it to be honest :-).

**Comment by axic** [comment]
4.8.7 or 4.8.9 ?

**Comment by ekpyron** [comment]
> 4.8.7 or 4.8.9 ?

Damn - commit message was wrong, I'll fix.

**Comment by ekpyron** [comment]
Builds are running (https://launchpad.net/~ethereum/+archive/ubuntu/cpp-build-deps/+packages).
Apparently ``eoan`` is already out of support and doesn't accept new uploads, so I'll remove it from the script.

**Review by leonardoalt — APPROVED** [review]

**Pull request merged**